integrationtests/tests/fixtures/mod.rs

```diff
@@ -945,16 +958,18 @@ impl FederationTest {
     /// it will wait forever
     pub async fn await_consensus_epochs(&self, epochs: usize) -> anyhow::Result<()> {
         for _ in 0..(epochs) {
-            for maybe_cancelled in join_all(
-                self.servers
-                    .iter()
-                    .map(|server| Self::consensus_epoch(server.clone(), Duration::from_millis(0))),
-            )
-            .await
-            {
-                maybe_cancelled?;
+            let mut task_group = TaskGroup::new();
+            for (i, server) in self.servers.iter().enumerate() {
+                let server = server.clone();
+                task_group
+                    .spawn(format!("server-{i}-consensu_epoch"), move |_| async {
+                        Self::consensus_epoch(server, Duration::from_millis(0)).await
```

Comment:
I don't think you need to spawn here? Wouldn't [`join_all`](https://docs.rs/futures/latest/futures/future/fn.join_all.html#) over an iterator of these futures also do the trick?

Reply:
`join_all` will not create separate tasks, so concurrency, but no parallelism.

Reply:
… which we need till other problems are fixed (moving heavy compute off tokio).

Reply:
But isn't the number of executor threads determined by the runtime anyway? Calling `tokio::spawn` does not mean tokio creates an actual thread afaik, just that this new task will be scheduled on available threads independent of the current task.

Reply:
Yes. And chances are other runtime threads have nothing better to do. :)

Reply:
Ah, I think I get your point now, even if compute is negligible/wouldn't *fill up* a thread it might lead to more latency if it is very bursty and happens at the same time.